crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -162,11 +162,7 @@ impl Cursor {
             return false;
         }
 
-        if !self.go_to_first_child() {
-            return self.go_to_next_non_descendent();
-        }
-
-        true
+        self.go_to_first_child() || self.go_to_next_non_descendent()
```

Comment:
I was gonna mumble about it but on the second thought it really is a lot clearer now, good thinking!